unlock-app/src/components/interface/checkout/index.tsx

```diff
@@ -46,13 +46,19 @@ export function CheckoutPage() {
     paywallConfig.referrer = referrerAddress
   }
 
-  const injectedProvider =
-    communication.providerAdapter || selectProvider(config)
-
   // Autoconnect, provider might change (we could receive the provider from the parent with a delay!)
   useEffect(() => {
-    authenticateWithProvider('DELEGATED_PROVIDER', injectedProvider)
-  }, [authenticateWithProvider, injectedProvider])
+    if (communication.providerAdapter) {
+      authenticateWithProvider(
+        'DELEGATED_PROVIDER',
+        communication.providerAdapter
+      )
+    }
+  }, [authenticateWithProvider, communication.providerAdapter])
```

Comment:
We only do the above for `communication.providerAdapter`... not for `selectProvider(config)`